plugins/flytekit-spark/flytekitplugins/spark/agent.py

```diff
@@ -70,10 +70,14 @@ async def async_get(self, context: grpc.ServicerContext, resource_meta: bytes) -
                 response = await resp.json()
 
         cur_state = PENDING
-        if response.get("state") and response["state"].get("result_state"):
-            cur_state = convert_to_flyte_state(response["state"]["result_state"])
-
-        return GetTaskResponse(resource=Resource(state=cur_state))
+        message = ""
+        if response.get("state"):
+            if response["state"].get("result_state"):
+                cur_state = convert_to_flyte_state(response["state"]["result_state"])
+            if response["state"].get("state_message"):
+                message = response["state"]["state_message"]
```

Comment:
nit: could we update it to
```suggestion
       state = response.get("state")
        if state:
            if state.get("result_state"):
                cur_state = convert_to_flyte_state(state["result_state"])
            if state.get("state_message"):
                message = state["state_message"]
```